### PR TITLE
Enable auto_reload_config only when Vault is enabled.

### PR DESCRIPTION
**Pull request by thisisnotashwin**
Changes proposed in this PR:
- Enable auto_reload_config only when Vault is enabled.

How I've tested this PR:
- Bats tests
How I expect reviewers to test this PR:
- 👀 

Checklist:
- [x] Tests added
- [x] CHANGELOG entry added 
  > HashiCorp engineers only, community PRs should not add a changelog entry.
  > Entries should use present tense (e.g. Add support for...)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -12,6 +12,7 @@ BUG FIXES:
   * Update create-federation-secret-job to look up the automatically generated gossip encryption key by the right name when global.name is unset or set to something other than consul. [[GH-1196](https://github.com/hashicorp/consul-k8s/pull/1196)]
   * Add Admin Partitions support to Sync Catalog **(Consul Enterprise only)**. [[GH-1180](https://github.com/hashicorp/consul-k8s/pull/1190)]
   * Correct webhook-cert-manager-clusterrole to utilize the web-cert-manager podsecuritypolicy rather than connect-injectors when `global.enablePodSecurityPolicies` is true. [[GH-1202](https://github.com/hashicorp/consul-k8s/pull/1202)]
+  * Enable Consul auto-reload-config only when Vault is enabled. [[GH-XXX](https://github.com/hashicorp/consul-k8s/pull/XXX)]
```
Comment (jmurret) [inline CHANGELOG.md:15]: Need to update the `XXX`

Comment (jmurret) [inline CHANGELOG.md:15]:
```suggestion
  * Enable Consul auto-reload-config only when Vault is enabled. [[GH-1213](https://github.com/hashicorp/consul-k8s/pull/1213)]
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,9 @@
 ## UNRELEASED
 
+BREAKING CHANGES:
+* Helm
+  * Requires Consul 1.12.0+. If you're not using the Vault integration then 0.43.1 will support Consul 1.11.x.
```
Comment (lkysow) [inline CHANGELOG.md:5]:
```suggestion
  * If you're using the Vault integration, requires Consul 1.12.0+.
```

**Review by lkysow — APPROVED** [review]
Just a suggestion to make the BC less scary.

**Pull request merged**